Experimental/DAG/luigi_test.py

```diff
@@ -0,0 +1,119 @@
+import os
+import re
+import subprocess
+from glob import glob
+
+import luigi
+
+# Let's define the relationships between the source files here. In practice
+# this information would come from Fab, perhaps it would be stored in
+# something like a sqlite database or similar
+DEPENDENCIES = {
+    "test_exec":
+        ["test_main.f90",],
+    "test_main.f90":
+        ["test.f90",],
+    }
```

Comment:
In an actual build `test_exec` would depend on `test_main.o`, not the source file. And `test_main.o` would depend on `test_main.f90` and `test.mod`.

Is that relevant to this example?

Reply:
You're right.  I did it this way to keep it simpler with less steps to worry about (and to be honest it was already becoming cumbersome even ignoring those extra layers of relationships between things)